sql/analyzer/hoist_filters.go

```diff
@@ -150,13 +141,56 @@ func recurseSubqueryForOuterFilters(n sql.Node, a *Analyzer, scope *plan.Scope)
 			return n, transform.SameTree, nil
 		}
 
+		if corr.Empty() {
+			// rootscope or equivalent, there is no benefit from hoisting
+			// we should materialize filters
+			defer func() {
```

Comment:
Don't do this, it's spooky action at a distance

You can edit the hoistFilters directly here if you need to,  or set a flag to be read in the final statements of the outer func